source/store/actions/dynamicFormData.ts

```diff
@@ -0,0 +1,106 @@
+import { Form } from '../../types/FormTypes';
+import { Case } from '../../types/CaseType';
+import { User } from '../../types/UserTypes';
+
+const addIds = (descriptor: string[], questionId: string, formId: string) =>
+  descriptor.map(s => {
+    if (s.split('.')[0] !== 'user' && questionId !== '') {
+      return `${formId}.${questionId}.${s}`;
+    }
+    if (s.split('.')[0] !== 'user' && questionId === '') {
+      return `${formId}.${s}`;
+    }
+    return s;
+  });
```

Comment:
Missing descriptive comment on this function. Hard to understand what the purpose of this function is. Get that it's adding questionId and formId to the descriptor. The function name should be more descriptive than addIds